action.yml

```diff
@@ -3,10 +3,14 @@ description: 'Setup your runner with Java'
 author: 'GitHub'
 inputs: 
   java-version:
-    description: 'The JDK version to make available on the path. Takes a whole or semver Jdk version, or 1.x syntax (e.g. 1.8 => Jdk 8.x)'
+    description: 'The Java version to make available on the path. Takes a whole or semver Java version, or 1.x syntax (e.g. 1.8 => Jdk 8.x)'
     required: true
+  java-package:
+    description: 'The package type (jre, jdk, jdk+fx)'
+    required: true
```

Comment:
Could you change this `false`, same thing for the architecture. They have default versions so they don't need to be  explicitly included in the YAML.

I tested it out with my own fork and it looks to be working as expected if you don't provide them. This is good because it won't break any existing workflows that use this action.
https://github.com/konradpabjan/Testing3/blob/master/.github/workflows/java.yml
https://github.com/konradpabjan/setup-java

I overall really like the direction of this. If there are any other packages in the future that users would like then it makes it easy to add it them here and no more extra input parameters will have to be added

Reply:
I was wondering about that. I chose to mimick the way the existing architecture input is defined, and for that one required is set to true, but the input provides a default. If we change one we should probably change both, and for consistency, we would also need to change the way they are consumed in e.g. https://github.com/foo-actions/setup-java/blob/master/src/setup-java.ts#L12 and https://github.com/foo-actions/setup-java/blob/master/lib/setup-java.js#L29

Reply:
If an input type has a default value then I'm leaning towards setting its required type to `false`. I think of what is in `action.yml` as the interface for whoever is using the action and separate from whatever is in `src/` consuming it. If an input has a default value and is false, then it makes it abundantly clear that the end user doesn't have to actually provide anything when using the action for that input type and they can use something like:

```
- uses: actions/setup-java@v1
  with:
    java-version: '9.0.4'
```

When an input is attempted to be read/consumed, its required type is set to false by default. In our case we have it set to `true` which will result in an exception being thrown if nothing is provided which is what we want to happen. https://github.com/actions/toolkit/blob/master/docs/specs/package-specs.md

Behind the scenes, a default value that we specify in `action.yml` creates an environment variable on the runner: https://help.github.com/en/actions/automating-your-workflow-with-github-actions/metadata-syntax-for-github-actions#inputs

If there is some error on the GitHub back-end that causes environment variables to not be created (highly unlikely) then an exception being thrown when trying to read the default input env variable will help identify that it's not an end-user error.

Reply:
I'm fine with either way for this, but should we change the architecture option the same way at the same time? (It seems wrong to have one do it one way and the other do it the other way)

Reply:
Yes, architecture as well (knock them both out), so it should be:
```
java-package:
    description: 'The package type (jre, jdk, jdk+fx)'
    required: false
    default: 'jdk'
architecture:
    description: 'The architecture (x86, x64) of the package.'
    required: false
    default: 'x64'
```

Reply:
Ok. And to be clear, leave https://github.com/foo-actions/setup-java/blob/master/src/setup-java.ts#L12 as it is?
Or do we need to change:
    `const javaPackage = core.getInput('java-package', {required: true});`
to something like:
    `const javaPackage = core.getInput('java-package', {required: false}) || 'jdk';`
?

Reply:
Yes, leave everything in `setup-java.ts` as is. The only changes are going to be in `action.yml`